src/js/components/Typeahead.js

```diff
@@ -0,0 +1,83 @@
+import classNames from 'classnames';
+import InnerTypeahead from 'mesosphere-react-typeahead';
+/* eslint-disable no-unused-vars */
+import React, {PropTypes} from 'react';
+/* eslint-enable no-unused-vars */
+
+import FilterInputText from './FilterInputText';
+
+class Typeahead extends FilterInputText {
+  handleInputChange(input) {
+    this.props.handleFilterChange(input);
+  }
+
+  handleClearInput() {
+    if (this.typeahead) {
+      this.typeahead.clear();
+    };
+  }
+
+  getInputField() {
+    let {
+      emptyLabel,
+      labelKey,
+      inverseStyle,
+      items,
+      onDropdownItemSelection,
+      placeholder,
+      selected
+    } = this.props;
+
+    if (selected === '') {
+      selected = [];
+    }
+
+    let classSet = classNames('typeahead', {
+      'inverse': inverseStyle
+    });
+
+    return (
+      <div className={classSet}>
+        <InnerTypeahead
+          emptyLabel={emptyLabel}
+          labelKey={labelKey}
+          onChange={onDropdownItemSelection}
+          onInputChange={this.handleInputChange}
+          options={items}
+          placeholder={placeholder}
+          ref={ref => { if (ref) { this.typeahead = ref.getInstance(); } }}
```

Comment:
Could we get a comment about why we need to use the `this.getInstance` function, and maybe a reference to here? https://github.com/mesosphere/react-typeahead#public-methods